### PR TITLE
chore: redux update

### LINKED ISSUE #4651: Dependencies (suite): react-redux, redux, redux-thunk, immer, @reduxjs/toolkit
Update `react-redux`, `redux`, `redux-thunk`, `immer`, `@reduxjs/toolkit`

Issues:

- update of `redux` breaks `type-check`
- update `react-redux` breaks `type-check` https://redux.js.org/tutorials/typescript-quick-start
- `redux-thunk` has types patch file which probably has to be changed
- bug fix https://github.com/trezor/trezor-suite/pull/4711#issue-1098109503
- `immer` breaks `type-check`
- `@reduxjs/toolkit` breaks `type-check`

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* updated  `react-redux`, `redux`, `redux-thunk`, `immer` and `@reduxjs/toolkit` to latest versions
* properly typed `useDispatch`
* removed the `redux-thunk` patch 🔪 
* some random files changed because of `lint --fix`

## Related Issue 

Resolve https://github.com/trezor/trezor-suite/issues/4651

### CODE REVIEW

File: packages/suite/src/hooks/suite/useDispatch.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { useDispatch as useReduxDisparch } from 'react-redux';
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useDispatch.ts]:
```suggestion
import { useDispatch as useReduxDispatch } from 'react-redux';
```

File: packages/suite/src/hooks/suite/useDispatch.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+import { ThunkDispatch } from 'redux-thunk';
+import { Action, AppState } from '@suite-types';
+
+export const useDispatch: () => ThunkDispatch<AppState, any, Action> = useReduxDisparch;
```
Comment (tomasklim) [inline packages/suite/src/hooks/suite/useDispatch.ts]:
```suggestion
export const useDispatch: () => ThunkDispatch<AppState, any, Action> = useReduxDispatch;
```

**Review by tomasklim — CHANGES_REQUESTED** [review]
Good job 🥳 

Please run `yarn dedupe`

I am getting those `type-check` errors 
```
    ✖  nx run @trezor/suite:type-check
       src/middlewares/suite/__tests__/buttonRequestMiddleware.test.ts(86,9): error TS2578: Unused '@ts-expect-error' directive.
       src/middlewares/suite/__tests__/buttonRequestMiddleware.test.ts(88,9): error TS2578: Unused '@ts-expect-error' directive.
```

**Comment by matejkriz** [comment]
@dahaca Have you checked this comment?
https://github.com/trezor/trezor-suite/pull/4711#issue-1098109503
> removed react-redux from suite-web package.json and pinned its version to 7.2.2. 7.2.3 contains a bug fix that is a breaking change for us - runtime looks fine BUT useDiscovery tests were relying on the old behavior and I am not 100% sure whether simply adjusting test fixtures is proper fix here.

**Comment by dahaca** [comment]
> @dahaca Have you checked this comment?
> 
> https://github.com/trezor/trezor-suite/pull/4711#issue-1098109503

No I haven't :( But I noticed something is up with the tests, will look into it.

File: packages/suite/src/hooks/suite/useDispatch.ts (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/hooks/suite/useDispatch.ts]: cd97cd0ec5fc50380d74d894c3853cd0bc5930f1

File: packages/suite/src/hooks/suite/useDispatch.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/hooks/suite/useDispatch.ts]: cd97cd0ec5fc50380d74d894c3853cd0bc5930f1

**Comment by dahaca** [comment]
@matejkriz @tomasklim test fixed e8cf055b2021e32810bf77f75047d10cecb38065

> Please run `yarn dedupe`

36a013f29005ceb60696c397e6c17302e86f352a

**Review by tomasklim — APPROVED** [review]
Good job!

you-did-good-job-meme

**Comment by Nodonisko** [comment]
We merged some new packages in meantime, please rebase and fix version also in them. Thanks.

**Pull request merged**